llama-index-core/llama_index/core/instrumentation/dispatcher.py

```diff
@@ -17,22 +18,33 @@
 import wrapt
 
 
-span_ctx = ContextVar("span_ctx", default={})
+# ContextVar's for managing active spans
+async_span_ctx = ContextVar(
+    "async_span_ctx", default=defaultdict(dict)
+)  # per thread >> async-task
+sync_span_ctx = ContextVar("sync_span_ctx", default={})  # per thread
```

Comment:
so to confirm
async_span_ctx is a dict of dicts, mapping thread -> async -> span

sync_span_ctx is a single dict, mapping thread -> span

Is that right? I wonder if we actually need both of these or not 🤔

Reply:
alright, was able to consolidate to one contextvar!